admin/yum.py

```diff
@@ -0,0 +1,192 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Effectful interface to RPM tools.
+"""
+
+import os
+from urlparse import urlparse
+
+from twisted.python.filepath import FilePath
+from characteristic import attributes
+from effect import Effect, sync_performer, TypeDispatcher
+from effect.do import do
+from subprocess import check_call
+from textwrap import dedent
+
+
+@attributes([
+    "source_repo",
+    "target_path",
+    "packages",
+])
+class DownloadPackagesFromRepository(object):
+    """
+    Download a given set of RPMs from a repository.
+
+    :ivar bytes source_repo: Location of repoisitory.
+    :ivar FilePath target_path: Directory to download packages to.
+    :ivar list packages: List of bytes, package names to download.
+    """
+
+
+@sync_performer
+def perform_download_packages_from_repository(dispatcher, intent):
+    """
+    See :class:`DownloadPackagesFromRepository`.
+    """
+    yum_repo_config = intent.target_path.child(b'build.repo')
+    yum_repo_config.setContent(dedent(b"""
+         [flocker]
+         name=flocker
+         baseurl=%s
+         """) % (intent.source_repo,))
+
+    check_call([
+        b'yum',
+        b'--config', yum_repo_config.path,
+        b'--disablerepo=*',
+        b'--enablerepo=flocker',
+        b'--quiet',
+        b'clean',
+        b'metadata'])
+
+    check_call([
+        b'yumdownloader',
+        b'--config', yum_repo_config.path,
+        b'--disablerepo=*',
+        b'--enablerepo=flocker',
+        b'--quiet',
+        b'--destdir', intent.target_path.path] + intent.packages)
```

Comment:
As mentioned elsewhere, we should probably just explicitly download the files we care about.